output/print_methods.go

```diff
@@ -2,59 +2,74 @@ package output
 
 import (
 	"fmt"
+	"net/http"
 	"strings"
 
 	"github.com/fatih/color"
 	"github.com/visola/go-http-cli/request"
 )
 
-type bodyBuffer struct {
-	*strings.Reader
-}
-
-func (bb *bodyBuffer) Close() error {
-	return nil
-}
-
 // PrintRequest outputs the http.Request
-func PrintRequest(request request.Request) error {
+func PrintRequest(request request.Request) {
 	boldGreen := color.New(color.Bold, color.FgGreen).PrintfFunc()
 	boldGreen("\n%s %s\n", request.Method, request.URL)
 
-	sentHeaderKeyColor := color.New(color.Bold, color.FgBlack).PrintfFunc()
-	sentHeaderValueColor := color.New(color.FgBlack).PrintfFunc()
+	printHeaders(request.Headers)
+	printCookies(request.Cookies)
+	printBody(request.Body, ">>")
+}
 
-	for headerName, values := range request.Headers {
-		sentHeaderKeyColor("%s:", headerName)
-		if len(values) > 1 {
-			for _, val := range values {
-				sentHeaderValueColor("\n  %s", val)
-			}
-			fmt.Println("")
-		} else {
-			sentHeaderValueColor(" %s\n", values[0])
+// PrintResponse outputs a http.Response
+func PrintResponse(response request.Response) {
+	if response.StatusCode < 300 {
+		color.Green("%s %s\n", response.Status, response.Protocol)
+	} else if response.StatusCode < 400 {
+		color.Yellow("%s %s\n", response.Status, response.Protocol)
+	} else {
+		color.Red("%s %s\n", response.Status, response.Protocol)
+	}
+
+	printHeaders(response.Headers)
+	printBody(response.Body, "<<")
+}
+
+func printBody(body string, linePrefix string) {
+	if body != "" {
+		bodyColor := color.New(color.Bold).PrintfFunc()
+		split := strings.Split(body, "\n")
+		for _, line := range split {
+			bodyColor("%s %s\n", linePrefix, line)
 		}
 	}
+}
 
-	if len(request.Cookies) > 0 {
+func printCookies(cookies []*http.Cookie) {
+	if len(cookies) > 0 {
 		sentCookieKeyColor := color.New(color.Bold, color.FgBlue).PrintfFunc()
 		sentCookieValueColor := color.New(color.FgBlue).PrintfFunc()
 
 		sentCookieKeyColor("Cookies:")
-		for _, cookie := range request.Cookies {
+		for _, cookie := range cookies {
 			sentCookieKeyColor("\n  %s: ", cookie.Name)
 			sentCookieValueColor("%s", cookie.Value)
 		}
 		fmt.Println("")
 	}
+}
 
-	if request.Body != "" {
-		split := strings.Split(request.Body, "\n")
-		for _, line := range split {
-			fmt.Printf(">> %s\n", line)
+func printHeaders(headers map[string][]string) {
+	sentHeaderKeyColor := color.New(color.Bold, color.FgBlack).PrintfFunc()
+	sentHeaderValueColor := color.New(color.FgBlack).PrintfFunc()
+
+	for headerName, values := range headers {
+		sentHeaderKeyColor("%s:", headerName)
+		if len(values) > 1 {
+			for _, val := range values {
+				sentHeaderValueColor("\n  %s", val)
+			}
+			fmt.Println("")
+		} else {
+			sentHeaderValueColor(" %s\n", values[0])
```

Comment:
what if `len(values) = 0` ?

Reply:
I think that's an edge case that will very rare, but I'll add some logic to handle it since it's supported by the protocol.